src/actions/actions.js

```diff
@@ -33,6 +36,23 @@ export function updateAppMessages(messageObj) {
   }
 }
 
+export function updateJsmdContent(text) {
+  return (dispatch) => {
+    const jsmdChunks = jsmdParser(text)
+    const reportChunks = jsmdChunks
+      .filter(c => c.chunkType === 'md' || c.chunkType === 'html')
+    dispatch({
+      // this dispatch really just forwards to the eval frame
+      type: 'UPDATE_MARKDOWN_CHUNKS',
+      reportChunks,
+    })
```

Comment:
Maybe there's something I'm missing, but I'm wondering if maybe we should consolidate UPDATE_MARKDOWN_CHUNKS into UPDATE_JSMD_CONTENT, and have the eval frame process the latter? I don't feel like one action really makes sense without the other. Maybe we should also parse out the report chunks in the reducer?

Reply:
this is admittedly a hack for now, but i think it's the right thing pending #1175 -- for now, this weird pattern is the "correct" way of sending the report chunks content to the eval frame without sending down additional data.